### PR TITLE
fix(server-kafka): Remove zookeeper and club-topicana-core from dependencies

### PR DESCRIPTION
**Pull request by christopher-cudennec**
Fixes security findings for:
* CVE-2019-20444
* CVE-2019-20445

### CODE REVIEW

File: sda-commons-server-kafka/README.md [hunk 1]
```diff
@@ -19,7 +19,7 @@ compile 'org.sdase.commons:sda-commons-server-kafka:<current-version>'
 
 | Group                        | Name                    | Version     | Description                  |
 |------------------------------|-------------------------|-------------|------------------------------|
-| `org.apache.kafka`           | `kafka-clients`         | 1.1.1       | Client API for Apache Kafka  |
+| `org.apache.kafka`           | `kafka-clients`         | 2.2.1       | Client API for Apache Kafka  |
 | `com.github.ftrossbach`      | `club-topicana-core`    | 0.1.0       | Helper for Topic description |
```
Comment (Fox32) [inline sda-commons-server-kafka/README.md:23]: Do we need this table? I think that will go out of date to quickly?

File: sda-commons-server-kafka/build.gradle [hunk 2]
```diff
@@ -10,21 +10,8 @@ dependencies {
 
   compile ('com.github.ftrossbach:club-topicana-core') {
     exclude group: 'org.slf4j', module: 'slf4j-log4j12'
-    exclude group: 'org.apache.zookeeper', module: 'zookeeper'
     exclude group: 'org.apache.kafka', module: 'kafka_2.12'
     exclude group: 'org.apache.kafka', module: 'kafka_2.11'
-    exclude group: 'org.apache.kafka', module: 'kafka-clients'
-  }
-
-  compile "org.apache.kafka:kafka-clients"
```
Comment (dhenneke) [inline sda-commons-server-kafka/build.gradle:19]: We should leave this here and don't use the transitive version from clup-topicana?

File: sda-commons-server-kafka/build.gradle [hunk 3]
```diff
@@ -10,21 +10,8 @@ dependencies {
 
   compile ('com.github.ftrossbach:club-topicana-core') {
     exclude group: 'org.slf4j', module: 'slf4j-log4j12'
-    exclude group: 'org.apache.zookeeper', module: 'zookeeper'
     exclude group: 'org.apache.kafka', module: 'kafka_2.12'
     exclude group: 'org.apache.kafka', module: 'kafka_2.11'
```
Comment (dhenneke) [inline sda-commons-server-kafka/build.gradle:14]: It should be enough to only exclude kafka_2.12 and kafka-clients. There is no dependency to the other noted dependencies (see https://search.maven.org/artifact/com.github.ftrossbach/club-topicana-core/0.1.0/jar).

**Review by dhenneke — COMMENTED** [review]
Removing zkclient and zookeeper should be OK as long as the tests are green.

File: sda-commons-server-kafka/build.gradle (hunk 2 above)
Comment (christopher-cudennec) [reply sda-commons-server-kafka/build.gradle:19]: I restored the dependency here as it is actively used in the module.

File: sda-commons-server-kafka/README.md (hunk 1 above)
Comment (christopher-cudennec) [reply sda-commons-server-kafka/README.md]: Not a fan either. I removed that section from the documentation.

File: sda-commons-server-kafka/build.gradle (hunk 3 above)
Comment (christopher-cudennec) [reply sda-commons-server-kafka/build.gradle]: Done as suggested. 👍

File: sda-commons-server-kafka/src/main/java/org/sdase/commons/server/kafka/topicana/TopicComparer.java [hunk 4]
```diff
@@ -36,7 +33,7 @@
 import org.sdase.commons.server.kafka.KafkaProperties;
 
 /**
- * This is a copy of the original {@link com.github.ftrossbach.club_topicana.core.TopicComparer}
```
Comment (Fox32) [inline sda-commons-server-kafka/src/main/java/org/sdase/commons/server/kafka/topicana/TopicComparer.java:39]: Don't we need to add the license to this file too?

File: sda-commons-server-kafka/src/main/java/org/sdase/commons/server/kafka/topicana/TopicConfigurationBuilder.java [hunk 5]
```diff
@@ -1,7 +1,5 @@
 package org.sdase.commons.server.kafka.topicana;
 
-import com.github.ftrossbach.club_topicana.core.ExpectedTopicConfiguration;
-
 public class TopicConfigurationBuilder {
```
Comment (Fox32) [inline sda-commons-server-kafka/src/main/java/org/sdase/commons/server/kafka/topicana/TopicConfigurationBuilder.java:3]: Is this file copied? If yes, we should add the license header

File: sda-commons-server-kafka/src/main/java/org/sdase/commons/server/kafka/topicana/TopicComparer.java (hunk 4 above)
Comment (dhenneke) [reply sda-commons-server-kafka/src/main/java/org/sdase/commons/server/kafka/topicana/TopicComparer.java:39]: Look at the start of this file :)

Comment (Fox32) [reply sda-commons-server-kafka/src/main/java/org/sdase/commons/server/kafka/topicana/TopicComparer.java:39]: We already have one, perfect

**Review by Fox32 — DISMISSED** [review]
👍 Let's see if it's green

File: sda-commons-server-kafka/src/main/java/org/sdase/commons/server/kafka/topicana/TopicConfigurationBuilder.java (hunk 5 above)
Comment (christopher-cudennec) [reply sda-commons-server-kafka/src/main/java/org/sdase/commons/server/kafka/topicana/TopicConfigurationBuilder.java:3]: This file was created by us. No license needed.

**Review by dhenneke — APPROVED** [review]

**Review by Fox32 — APPROVED** [review]
👍

**Pull request merged**